generic_scripts/nlk.py

```diff
@@ -0,0 +1,167 @@
+        return B
+
+    def include_pos_error(self, sigma):
+        self.error = _np.random.normal(loc=0, scale=sigma)
+        self.pos = self.pos + self.error
+
+    @property
+    def pos(self):
+        return self._pos
+
+    @pos.setter
+    def pos(self, position):
+        self._pos = position
+
+    @property
+    def curr(self):
+        return self._curr
+
+    @curr.setter
+    def curr(self, current):
+        self._curr = current
+
+
+class NLK:
+    """."""
+    def __init__(self, curr=1850, errors=False, sigma_errors=1e-5):
+        # Setting wire positions
+        wire_positions = []
+        wire_positions.append(_np.array([7.0,  5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([10.,  5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([-7.,  5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([-10,  5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([7.0, -5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([10., -5.85])[:, None]*1e-3)
+        wire_positions.append(_np.array([-7., -5.21])[:, None]*1e-3)
+        wire_positions.append(_np.array([-10, -5.85])[:, None]*1e-3)
+        self._positions = wire_positions
+
+        # Setting wire currents
+        currents = []
+        for i in range(8):
+            if i % 2:
+                currents.append(-curr)
+            else:
+                currents.append(curr)
+        self._currents = currents
+
+        # Creating wires
+        self._wires = []
+        for i in range(8):
+            pos = wire_positions[i]
+            curr = currents[i]
+            wire = Wire(pos, curr)
+            if errors:
+                wire.include_pos_error(sigma=sigma_errors)
+            self._wires.append(wire)
+
+    @property
+    def wires(self):
+        return self._wires
+
+    @wires.setter
+    def wires(self, wires_objects):
+        self._wires = wires_objects
+
+    @property
+    def positions(self):
+        "Return positions of all wires"
+        return self._positions
+
+    @positions.setter
+    def positions(self, wires_positions):
+        self._positions = wires_positions
+
+    @property
+    def currents(self):
+        "Return currents of all wires"
+        return self._currents
+
+    @currents.setter
+    def currents(self, wires_currents):
+        self._currents = wires_currents
+
+    def Bfield(self, x, y):
+        field = _np.zeros([2, 1])
+        for wire in self.wires:
+            field += wire.Bfield(x, y)
+        return field
+
+    def include_pos_error(self, sigma_errors):
+        for wire in self.wires:
+            wire.include_pos_error(sigma_errors)
+
+    def nlk_profile(self):
+        "NLK vertical field at y=0 for x ∈ [-12, 12] mm."
+        x_space = _np.linspace(-12, 12)*1e-3
+        fieldy = _np.array([self.Bfield(x, 0)[1] for x in x_space])
+        return x_space, fieldy
+
+
+def polyfit(x, y, n):
```

Comment:
ok, it is nice to implement a polynomial fitting by yourself, but I think `numpy.polynomial.polynomial.polyfit` optimized.